test/e2e/framework/client.go

```diff
@@ -72,13 +86,35 @@ func (cl *ClusterClient) ValidateObject(ctx context.Context, obj client.Object,
 	return nil
 }
 
+// ValidateObjectNotFound verifies the specified object does not exist.
+func (cl *ClusterClient) ValidateObjectNotFound(ctx context.Context, obj client.Object) error {
+	cl.Helper()
+	nn := types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}
+	cl.Logf("ValidateObjectNotFound %T (%s)", obj, nn.String())
+	err := cl.client.Get(ctx, nn, obj)
+	if err == nil { // object still exists - error
+		return fmt.Errorf("ValidateObjectNotFound %T (%s): object still exists",
+			obj, nn.String())
+	} else if !k8serrors.IsNotFound(err) { // unexpected error
+		return fmt.Errorf("ValidateObjectNotFound %T (%s): %w",
```

Comment:
nit: The errors from ValidateObjectNotFound are not exposed here. They are returned but not logged in the WaitForObjectNotFound loop, so we can't see why it might be failing intermittently. It might be helpful to log validation err.

Reply:
I see, this seems fine then!